UnityProject/Assets/Scripts/Player/Huggable.cs

```diff
@@ -1,17 +1,37 @@
+using System.Threading.Tasks;
 using HealthV2;
 using Messages.Server.SoundMessages;
 using UnityEngine;
+using System.Collections.Generic;
 
 
 /// <summary>
 /// Allows an object to be hugged by a player.
 /// </summary>
-public class Huggable : MonoBehaviour, ICheckedInteractable<HandApply>
+public class Huggable : MonoBehaviour, ICheckedInteractable<HandApply>, ICooldown
 {
 	private HandApply interaction;
 	private string performerName;
 	private string targetName;
 
+	/// <summary>
+	/// Time in-between gib checks for tail pulling In milliseconds
+	/// </summary>
+	private readonly float tailPullJudgementCooldown = 15;
+
+	public float DefaultTime { get; set; }
```

Comment:
you don't need the field above, use this property instead.
```suggestion
	public float DefaultTime {  get;  } = 15;
```

Reply:
done!